newton/_src/sim/builder.py

```
@@ -483,6 +485,194 @@ def __init__(self, up_axis: AxisType = Axis.Z, gravity: float = -9.81):
+                dof_end = self.joint_qd_start[joint_index + 1]
+            else:
+                dof_end = self.joint_dof_count
+
+            dof_count = dof_end - dof_start
+
+            for attr_name, value in dof_attrs.items():
+                # DOF attributes must always be lists
+                if not isinstance(value, (list, tuple)):
+                    raise TypeError(
+                        f"DOF attribute '{attr_name}' must be a list with length equal to joint DOF count ({dof_count})"
+                    )
```

Ruff (0.14.0)

Avoid specifying long messages outside the exception class

(TRY003)